src/Providers/Maps/Google/Features/Shape.ts

```diff
@@ -0,0 +1,144 @@
+            this._map = OutSystems.Maps.MapAPI.MapManager.GetMapById(mapId);
+            // Set the default return message to prevent different else if's
+            this._returnObjSuccess = false;
+            this._returnObjCode =
+                OSFramework.Maps.Enum.ErrorCodes.CFG_InvalidMapId;
+
+            // Check if map exists
+            if (this._map) {
+                this._isInsideShape = false;
+
+                // Convert to JSON the marker location before to convert to google format
+                this._markerCoordinates = JSON.parse(pointCoordinates);
+                this._markerCoordinates = new google.maps.LatLng(
+                    this._markerCoordinates.Lat,
+                    this._markerCoordinates.Lng
+                );
+
+                // Check if marker is inside shape based on shape element
+                if (shapeId) {
+                    this._shape =
+                        OutSystems.Maps.MapAPI.ShapeManager.GetShapeById(
+                            shapeId
+                        );
+
+                    // Check if shape contains marker based on shape type
+                    switch (this._shape.type) {
+                        case OSFramework.Maps.Enum.ShapeType.Polyline:
+                            // The Polyline is an unsupported shape to use the ContainsLocation API
+                            this._returnObjCode =
+                                OSFramework.Maps.Enum.Unsupported.code;
+                            this._returnObjMessage =
+                                OSFramework.Maps.Enum.Unsupported.message;
+                            break;
+                        case OSFramework.Maps.Enum.ShapeType.Rectangle:
+                            this._shapeRectangle();
+                            break;
+                        case OSFramework.Maps.Enum.ShapeType.Circle:
+                            this._shapeCircle();
+                            break;
+
+                        default: {
+                            this._shapeDefault();
+                        }
+                    }
+                } else {
+                    const shapeCoordinatesList = JSON.parse(
+                        coordinatesList.toLowerCase()
+                    );
+
+                    // To create a shape we need at least 3 coordinates
+                    if (shapeCoordinatesList.length >= 3) {
+                        const newShape = new google.maps.Polygon({
+                            paths: shapeCoordinatesList
+                        });
+
+                        this._isInsideShape =
+                            google.maps.geometry.poly.containsLocation(
+                                this._markerCoordinates,
+                                newShape
+                            );
+                    }
+                }
+            }
+
+            // Set the return message to expose values
+            const returnMessage = {
```

Comment:
I think you can return directly in this line, without the const

Reply:
Done